stable/spark-history-server/templates/deployment.yaml

```diff
@@ -60,31 +69,48 @@ spec:
           httpGet:
             path: /
             port: historyport
-        volumeMounts:
         {{- if .Values.pvc.enablePVC }}
+        volumeMounts:
         - name: data
           mountPath: /mnt/{{ .Values.pvc.eventsDir }}
         {{- else if .Values.gcs.enableGCS }}
+        volumeMounts:
         - name: secrets-volume
           mountPath: /etc/secrets
+        {{- else if .Values.s3.enableS3 }}
+        {{- if (not .Values.s3.enableIAM) }}
+        volumeMounts:
+        - name: secrets-volume
+          mountPath: /etc/secrets
+        {{- end}}
         {{- else }}
+        volumeMounts:
         - name: core-site
           mountPath: /etc/hadoop/core-site.xml
           subPath: core-site.xml
         - name: hdfs-site
           mountPath: /etc/hadoop/hdfs-site.xml
           subPath: hdfs-site.xml
         {{- end }}
-      volumes:
```

Comment:
Similar to the `volumeMounts` above, leaving `volumes` here would save you from repeat it below.

Reply:
The same reasoning for the `volumeMounts` comment above.